python/nammu/controller/AtfAreaController.py

```diff
@@ -186,13 +186,14 @@ def restore_highlight(self):
 
     def getPositionFromLine(self, text, line_num):
         '''
-        Given a block of text and a line number, return a tuple containing
-        the caret position at the start and end of the given line.
+        Given a block of text and a line number, return the caret position
+        at the start of the given line.
         '''
-        compiled = re.compile(r"\n")
-        textiter = compiled.finditer(text)
-        if line_num != 1:
-            pos = [m.start() for m in textiter][line_num - 2:line_num]
-        else:
-            pos = [0, textiter.next().start()]
+        pos = 0
```

Comment:
You don't need to run this line every time since it's most likely it'll get overwritten later. I'd put it inside an `else`. It wouldn't matter too much in general since this line is not doing much, but I think it's being called from the Syntax Highlighter every time there's a key pressed. Any execution time we can save on that would be good! (It might be negligible in this case, but good to keep in mind).